src/Command.js

```diff
@@ -33,6 +35,31 @@ export default class Command {
   }
 
   runValidations() {
+    if (!NpmUtilities.getWhoIAm()) {
+      this.logger.warn("You're not logged in; you may not be able to perform some commands.");
+    }
+
+    const lernaRegistry = ConfigUtilities.read(this.repository.rootPath).registry;
+    const npmRegistry = NpmUtilities.getConfig("registry");
+    if (lernaRegistry) {
+      if (npmRegistry !== lernaRegistry) {
+        this.logger.warn(
+          "Your lerna.json specifies that you should use the registry " +
+          `${lernaRegistry}.\nTo change npm to use ${lernaRegistry} ` +
+          "instead, run `npm config set registry ${lernaRegistry}`."
+        );
+      }
+    } else {
+      if (npmRegistry !== "http://registry.npmjs.com/") {
```

Comment:
As mentioned in https://github.com/lerna/lerna/pull/485, this should be `https://registry.npmjs.org/`.

(it's very confusing to "layer" PRs like this. As far as I can tell only `NpmUtilities.getConfig()` is necessary in the other PR, and it's very noisy to have a bunch of duplicate changes in both PRs...)